### ISSUE TITLE
Keyword "else" must not be followed by a newline

### ISSUE THREAD

**Issue by mhart** [eslint/eslint#3226]
The fix for #3198 has created a quirk in the way the `else` keyword is handled.

`"space-after-keywords": [2, "always"]` with the following code:

```js
if (a == null)
  doThing()
else
  doOtherThing()
```

Results in the error: `Keyword "else" must not be followed by a newline`

I believe this rule of strictly being a space is really intended to be applied to keywords where the next token is `(` or `{` – as in `if(something)` or `else{ whatever() }`

**Comment by mhart** [comment]
Actually, now that I say that, perhaps `{` is a case where a newline could be fine – but `(` is not

**Comment by btmills** [comment]
`{` should be covered by brace-style, however that's configured.

**Comment by aparajita** [comment]
For those of us who don't enforce braces, the "fix" to #3198 totally broke this rule with `else`.

**Comment by nzakas** [comment]
Yeah, seems like this is the wrong warning. Really, we shouldn't warn about a newline at all, we should just ignore it.

**Comment by btmills** [comment]
Wouldn't that just revert to the pre-#3198 behavior?

**Comment by nzakas** [comment]
Not exactly. If a space is expected but a newline is found, that seems like it should be an error?

**Comment by btmills** [comment]
It looks like this rule needs a little more work. Consider:

```js
/* eslint space-after-keywords: [2, "never"] */
if(a) b();
else c();
```

It will earn about the space between `else` and `c`, but that space is required.

**Comment by mhart** [comment]
It seems to me that `else` should just be excluded from this rule, no? You always have to have a space of some sort after else don't you?

**Comment by mhart** [comment]
Oh, unless it's a brace, but @btmills you [seemed to indicate](https://github.com/eslint/eslint/issues/3226#issuecomment-126799575) that should be covered by other rules.

**Comment by btmills** [comment]
Braces on new lines (or not) are covered, but not the space between else and the brace

**Comment by GianlucaGuarini** [comment]
[we](https://github.com/riot/riot) can not upgrade eslint because of this issue. Is there any milestone assigned to this issue?

**Comment by joezimjs** [comment]
Just curious if there is an ETA on releasing this fix? Thanks
